source/languageHandler.py

```diff
@@ -79,20 +115,90 @@ def getLanguageDescription(language):
 		desc=buf.value
 	if not desc:
 		#Some hard-coded descriptions where we know the language fails on various configurations.
-		desc={
-			# Translators: The name of a language supported by NVDA.
-			"an":pgettext("languageName","Aragonese"),
-			# Translators: The name of a language supported by NVDA.
-			"ckb":pgettext("languageName","Central Kurdish"),
-			# Translators: The name of a language supported by NVDA.
-			"kmr":pgettext("languageName","Northern Kurdish"),
-			# Translators: The name of a language supported by NVDA.
-			"my":pgettext("languageName","Burmese"),
-			# Translators: The name of a language supported by NVDA.
-			"so":pgettext("languageName","Somali"),
-		}.get(language,None)
+		# Imported lazily since langs description are translatable
+		# and `languageHandler` is responsible for setting the translation.
+		import localesData
+		desc = localesData.LANG_NAMES_TO_LOCALIZED_DESCS.get(language, None)
 	return desc
 
+
+def englishLanguageNameFromNVDALocale(localeName: str) -> Optional[str]:
+	"""Returns either English name of the given language  using `GetLocaleInfoEx` or None
+	if the given locale is not known to Windows."""
+	localeName = normalizeLocaleForWin32(localeName)
+	buffLength = winKernel.kernel32.GetLocaleInfoEx(localeName, LOCALE_SENGLISHLANGUAGENAME, None, 0)
+	if buffLength:
+		buf = ctypes.create_unicode_buffer(buffLength)
+		winKernel.kernel32.GetLocaleInfoEx(localeName, LOCALE_SENGLISHLANGUAGENAME, buf, buffLength)
+		langName = buf.value
+		if "Unknown" in langName:
+			# Windows 10 returns 'Unknown' for locales not known to Windows
+			# even though documentation states that in case of an unknown locale 0 is returned.
+			return None

```

Comment:
Could you add a test to confirm None is returned in this case?

Reply:
This is being tested by `test_noLangNameFromUnknownLocale`. Note that this test is a general one but that is because it is impossible to know in advance if the version of Windows on which it is executed returns 0 or 'unknown' for unknown locales so it merely checks if `None` is returned for unknown locale. Creating tests which ensures that both these paths results in `None` would be rather difficult, the only way I can thing of at present would be to:
- Create a wrapper around invocation of `GetLocaleInfoEx(localeName, LOCALE_SENGLISHLANGUAGENAME, None, 0)` which returns whatever the underlying system returns without any validation of the result
- Then create two unit tests in which this wrapper is mocked so that it return ''Unknown' for one test and 0 for the second.

I'm not convinced this is worth the trouble.

Reply:
Thanks, this justification is more than sufficient.